api/namex/services/name_request/name_request.py

```diff
@@ -140,24 +140,38 @@ def update_request_submit_count(cls, name_request):
         return name_request
 
     @classmethod
-    def extend_expiry_date(cls, name_request, start_date=None, days=56):
+    def get_expiry_days(cls, name_request):
+        """
+        returns expiry days of an NR.
+        """
+        if name_request.request_action_cd in [RequestAction.REH.value, RequestAction.REN.value]:
+            expires_days = ExpiryDays.NAME_REQUEST_REH_REN_LIFESPAN_DAYS.value
+        else:
+            expires_days = ExpiryDays.NAME_REQUEST_LIFESPAN_DAYS.value
+
+        return expires_days
+
+    @classmethod
+    def extend_expiry_date(cls, name_request, start_date=None):
         """
         Extends/sets the expiry date of an NR.
 
         Default: extends the expiry date to 56 days date at 11:59pm.
         """
         start_datetime = start_date if start_date else datetime.utcnow()
         try:
+            expiry_days = cls.get_expiry_days(name_request)
+           
             name_request.expirationDate = cls.create_expiry_date(
                 start=start_datetime,
-                expires_in_days=days
+                expires_in_days=expiry_days
             )
         except Exception as err:
             raise ExtendExpiryDateError(err)
 
         return name_request
 
-    def map_request_data(self, name_request, map_attrs=False):
+    def map_request_data(cls, self, name_request, map_attrs=False):
```

Comment:
This needs to be decorated as a class method
```suggestion
    @classmethod
    def map_request_data(cls, self, name_request, map_attrs=False):
```
You can accept the suggestion here and it will change the code.